From 911cbd384fc833f1a02796e7332edc698d10dcd1 Mon Sep 17 00:00:00 2001
From: Arien Judge <arien.judge@morsemicro.com>
Date: Thu, 26 Jun 2025 12:30:06 +1000
Subject: [PATCH 10/15] imx6ul-var-dart: add support for SPI attached Morse
 Micro MM6108

uses pins on J11 and J7 of the DART-6UL-Customboard to connect to a
Morse Micro MF08651
---
 .../imx/imx6ull-var-dart-6ulcustomboard.dtsi  | 97 +++++++++++++++++++
 1 file changed, 97 insertions(+)

diff --git a/arch/arm/boot/dts/nxp/imx/imx6ull-var-dart-6ulcustomboard.dtsi b/arch/arm/boot/dts/nxp/imx/imx6ull-var-dart-6ulcustomboard.dtsi
index f4540c5951bd..7a5af5f89164 100644
--- a/arch/arm/boot/dts/nxp/imx/imx6ull-var-dart-6ulcustomboard.dtsi
+++ b/arch/arm/boot/dts/nxp/imx/imx6ull-var-dart-6ulcustomboard.dtsi
@@ -15,3 +15,100 @@ MX6ULL_PAD_SNVS_TAMPER7__GPIO5_IO07	0x1b0b0 /* RTC alarm IRQ */
 		>;
 	};
 };
+
+
+&iomuxc {
+
+/*
+ *    MOSI J11.8  MX6UL_PAD_CSI_DATA06__ECSPI1_MOSI
+ *    MISO J11.6  MX6UL_PAD_CSI_DATA07__ECSPI1_MISO
+ *    SCK  J11.2  MX6UL_PAD_CSI_DATA04__ECSPI1_SCLK
+ *    CS   J11.4  MX6UL_PAD_CSI_DATA05__GPIO4_IO26
+ *    RST  J11.9  MX6UL_PAD_CSI_DATA00__GPIO4_IO21
+ *    IRQ  J6.5   MX6UL_PAD_CSI_DATA01__GPIO4_IO22
+ *    BSY  J6.7   MX6UL_PAD_CSI_DATA02__GPIO4_IO23
+ *    WKE  J12.8  MX6UL_PAD_CSI_MCLK__GPIO4_IO17
+ *
+ */
+
+/*
+PAD_CTL_HYS                     (1 << 16)
+PAD_CTL_PUS_100K_DOWN           (0 << 14)
+PAD_CTL_PUS_47K_UP              (1 << 14)
+PAD_CTL_PUS_100K_UP             (2 << 14)
+PAD_CTL_PUS_22K_UP              (3 << 14)
+PAD_CTL_PUE                     (1 << 13)
+PAD_CTL_PKE                     (1 << 12)
+PAD_CTL_ODE                     (1 << 11)
+PAD_CTL_SPEED_LOW               (0 << 6)
+PAD_CTL_SPEED_MED               (1 << 6)
+PAD_CTL_SPEED_HIGH              (3 << 6)
+PAD_CTL_DSE_DISABLE             (0 << 3)
+PAD_CTL_DSE_260ohm              (1 << 3)
+PAD_CTL_DSE_130ohm              (2 << 3)
+PAD_CTL_DSE_87ohm               (3 << 3)
+PAD_CTL_DSE_65ohm               (4 << 3)
+PAD_CTL_DSE_52ohm               (5 << 3)
+PAD_CTL_DSE_43ohm               (6 << 3)
+PAD_CTL_DSE_37ohm               (7 << 3)
+PAD_CTL_SRE_FAST                (1 << 0)
+PAD_CTL_SRE_SLOW                (0 << 0)
+*/
+
+
+/*
+			MX6UL_PAD_CSI_DATA07__ECSPI1_MISO       0x17040
+			MX6UL_PAD_CSI_DATA06__ECSPI1_MOSI       0x17059
+			MX6UL_PAD_CSI_DATA04__ECSPI1_SCLK       0x17059
+			MX6UL_PAD_CSI_DATA05__GPIO4_IO26        0x17059
+
+			MX6UL_PAD_CSI_DATA00__GPIO4_IO21        0x1b0b1
+			MX6UL_PAD_CSI_DATA01__GPIO4_IO22        0x1b0b1
+			MX6UL_PAD_CSI_DATA02__GPIO4_IO23        0x1b0b1
+			MX6UL_PAD_CSI_MCLK__GPIO4_IO17          0x100b1
+
+*/
+
+	pinctrl_ecspi1_1: ecspi1grp {
+		fsl,pins = <
+			MX6UL_PAD_CSI_DATA07__ECSPI1_MISO       0x100b1
+			MX6UL_PAD_CSI_DATA06__ECSPI1_MOSI       0x100b1
+			MX6UL_PAD_CSI_DATA04__ECSPI1_SCLK       0x100b1
+			MX6UL_PAD_CSI_DATA05__ECSPI1_SS0        0x100b1
+			/* MX6UL_PAD_CSI_DATA05__GPIO4_IO26        0x100b1 */
+
+		>;
+	};
+
+	pinctrl_mm6108: mm6108grp {
+		fsl,pins = <
+			MX6UL_PAD_CSI_DATA00__GPIO4_IO21        0x100b9
+			MX6UL_PAD_CSI_DATA01__GPIO4_IO22        0x100b9
+			MX6UL_PAD_CSI_DATA02__GPIO4_IO23        0x100b9
+			MX6UL_PAD_CSI_MCLK__GPIO4_IO17          0x100b9
+		>;
+	};
+};
+
+&ecspi1 {
+	// fsl,espi-num-chipselects = <1>;
+	// fsl,spi-num-chipselects = <1>;
+	// num-cs = <1>;
+	// cs-gpios = <&gpio4 26 1>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_ecspi1_1>;
+	status = "okay";
+
+	mm6108: mm6108@0 {
+		compatible = "morse,mm610x-spi";
+		spi-max-frequency = <50000000>;
+		reg = <0>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_mm6108>;
+		reset-gpios = <&gpio4 21 0>;
+		power-gpios = <&gpio4 23 0>,
+		              <&gpio4 17 0>;
+		spi-irq-gpios = <&gpio4 22 0>;
+		status = "okay";
+	};
+};
\ No newline at end of file
-- 
2.25.1

